### ISSUE TITLE
Deprecated: Creation of dynamic properties in ElementNameResolver.php on line 174

### ISSUE THREAD

**Issue by eurosat7** [phpDocumentor/Reflection#529]
Ran phpDocumentor as phpdoc.phar with version v3.4.3 on a php 8.3 code base with high php error reporting.

```
> ./phpdoc.phar --target=documentation --directory=src
phpDocumentor v3.4.3

Parsing files

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Namespace_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Class_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Namespace_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Class_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Const_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Namespace_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Class_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Const_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Namespace_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Class_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Namespace_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\Class_::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Deprecated: Creation of dynamic property PhpParser\Node\Stmt\ClassMethod::$fqsen is deprecated in phar:///var/www/html/phpdoc.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/NodeVisitor/ElementNameResolver.php on line 174

Applying transformations (can take a while)

All done in 0 seconds!
```

**Comment by eurosat7** [comment]
Adding properties in nikic/php-parser to theese classes would solve it. 
```
PhpParser\Node\Const_::$fqsen 
PhpParser\Node\Stmt\ClassMethod::$fqsen 
PhpParser\Node\Stmt\Class_::$fqsen 
PhpParser\Node\Stmt\Namespace_::$fqsen 
```
But is that a solution?

**Comment by smaddock** [comment]
I believe this was resolved with 5644252

**Comment by jaapio** [comment]
Solved in our new v6 release
